core/services/ardupilot_manager/firmware_upload/FirmwareUpload.py

```diff
@@ -0,0 +1,86 @@
+import pathlib
+import shutil
+import subprocess
+import threading
+
+from loguru import logger
+
+
+class FirmwareUploadFail(RuntimeError):
+    pass
+
+
+class UploadToolNotFound(RuntimeError):
+    pass
+
+
+class FirmwareUpload:
+    def __init__(self) -> None:
+        self._autopilot_port: pathlib.Path = pathlib.Path("/dev/autopilot")
```

Comment:
is this not used? I couldn't find any uses of this, but pylint didn't complain about it.

Reply:
Good catch.
Pylint is not complaining because there's the `set_autopilot_port` method.
This variable should be on the subprocess call, but I mistakenly removed on the last rebase.
Fixed.